docs/docs/lists-conditional-rendering.md

```diff
@@ -0,0 +1,264 @@
+
+This follows the concept of "separation of concerns" by allowing each component to manage how it should be rendered.
+
+### Element Variables
+
+Given the current state of your application you can assign elements as the value of variables, then only render the needed element.
+
+In the example below we have a stateful component named `<LoginControl />`. We use a boolean value `this.state.loggedIn` to track if the user is logged in. If logged in, we want to render `<LogoutButton />`. If logged out, we want to render `<LoginButton />`. Depending on the value of `this.state.loggedIn`, we use `loginButton` to store the element to be rendered.
+
+The highlighted lines below show where `loginControlButton` is created and set to a default value of "`<LoginButton />`", then where it conditionally gets changed depending on current state, and finally where `{loginControlButton}` is rendered.
+
+```javascript{21,23,27}
+function LogoutButton(props) {
+  return <button onClick={props.click}>Logout</button>
+}
+function LoginButton(props) {
+  return <button onClick={props.click}>Login</button>
+}
+class LoginControl extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {loggedIn: true}
+    this.login = this.login.bind(this);
+    this.logout = this.logout.bind(this);
+  }
+  login() {
+    this.setState({loggedIn: true,});
+  }
+  logout() {
+    this.setState({loggedIn: false});
+  }
+  render() {
+    var loginControlButton = <LoginButton click={this.login} />;
+    if (this.state.loggedIn) {
+      loginControlButton = <LogoutButton click={this.logout} />;
+    }
+    return (
+      <div>
+        {loginControlButton}
+      </div>
+    );
+  }
+}
+
+```
+
+<a target="_blank" href="https://codepen.io/ericnakagawa/pen/Egpdrz?editors=0010">Try it out on Codepen.</a>
+
+
+### Prevent Component Rendering
+
+In some cases, you will not want a component to render. To prevent components from rendering, return `null` or `false` from the `render()` function.
+
+In the example below, a `<Page />` component renders a child `<WarningBanner />` element and a button. The button calls `this.toggleWarning` to switch the value of `this.state.showWarning`. If the `this.state.showWarning` is false, then nothing will be rendered.
+
+The highlighted lines below show where the variable `warningBanner` is set to a default value of null and then where it is rendered in `return()`.
+
+```javascript{17,23}
+function WarningBanner(props) {
+  return <div className="warning">Warning!</div>
+}
+class Page extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {showWarning: true, toggleText: "hide"}
+    this.toggleWarning = this.toggleWarning.bind(this);
+  }
+  toggleWarning() {
+    this.setState({
+      showWarning: !this.state.showWarning,
+      toggleText: !this.state.showWarning?"hide":"show"
+    });
```

Comment:
toggleText should not be in state. As mentioned in "Lifting State Up", state should be minimal. Everything that can be computed from it in render should be. This is why showWarning is enough, and toggleText should be calculated in render from it.